src/Pages/components/ContactForm.jsx

```diff
@@ -0,0 +1,84 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+
+const DEFAULS_CONTACT = {
+  contactId: 1,
+  firstName: "",
+  lastName: "",
+  city: "",
+};
+
+export default function ContactForm({ submitCallBack, contact }) {
+  const [formData, setFormData] = useState(
+    contact ? contact : { ...DEFAULS_CONTACT }
+  );
+  const [error, setError] = useState("");
+  const [creating, setCreating] = useState(false);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    submitCallBack(formData).then((error) => {
+      if (error) {
+        setError(error);
+      } else {
+        setCreating(true);
```

Comment:
You're setting the creating state to true after submission, which is intended to indicate that a contact has been created. However, this state doesn't get reset after displaying the message, which could confuse users if they go to create another contact without refreshing the page. Consider resetting the form and creating state upon successful submission